### ISSUE TITLE
Next suggestion isn't read

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#174362]
Discovered by @jooyoungseo

1. Trigger an inline suggestion
2. Trigger the hover
3. Trigger the next suggestion using the keybinding
4. 🐛 the screen reader doesn't read the next suggestion. The hover has to be closed and reopened for it to be read

cc @isidorn

**Comment by isidorn** [comment]
@meganrogge this for me is the biggest issue with the Copilot experience. Can we assign it to the March milestone so we do not lose track of it?

**Comment by meganrogge** [comment]
After exploring a bit, this is a duplicate of the request to have the inline suggestion read IMO.

**Comment by meganrogge** [comment]
We shouldn't need to use the hover to read the suggestion and to do so would be hacky I think

**Comment by meganrogge** [comment]
I investigated this and seems like the right fix would be:

1. When the suggestion changes, check if the hover is showing. 
2. If it is, update the text content (and screen reader content)
3. If it's not, if screen reader mode is enabled, trigger the hover to show with the updated suggestion. 

Thoughts @hediet ?

**Comment by meganrogge** [comment]
I actually think we shouldn't need to do anything with the hover here. We should be providing the screen reader with the inline suggestion as it changes. Work started on`merogge/ariaAlert2`, though `alert` isn't working for me at all on macOS with VoiceOver

**Comment by isidorn** [comment]
@meganrogge I think for this issue we should make sure that the hover experience is good. So if the user decided to show the hover, and transitions to the next suggestion the hover should update and the new one should get read - and not by `alert`.

So I would first investigate the current focus movement when hover is shown. Does the focus stay in the hover when a new suggestion gets rendered? If we fix the hover, does the right thing get read?

Apart from this effort we can look into using alerts for reading suggestions in general (unrelated to the hover). This should be behind a setting.

**Comment by meganrogge** [comment]
Yes I agree @isidorn 👍🏼 we want to make sure the hover works as expected and I reopened https://github.com/microsoft/vscode/issues/167348

**Comment by meganrogge** [comment]
Now that we have `editor.screenReaderAnnounceInlineSuggestion`, the accessibility part of this is fixed as a screen reader user will hear the next suggestion when the ghost text is updated. Leaving the hover update issue with Henning

**Comment by isidorn** [comment]
@meganrogge based on your comment I am assigning to Henning

@hediet I have removed the milestone so we plan for this in one of the future releases. I will also discuss with Alex.

**Comment by hediet** [comment]
I verified that this works now.
